### PR TITLE
8518/update order creation UI test

### PR DESCRIPTION
**Pull request by samiuelson**
Closes: #8518 

### Description
This PR updates UI tests related to Order creation after switching to using `ProductSelectorFragment` for products multiselection.

🚨 **Important:** Should be reviewed and merged after https://github.com/woocommerce/woocommerce-android/pull/8475 is merged.

### CODE REVIEW

**Comment by samiuelson** [comment]
Hey @jostnes! I've updated the UI tests of the Order creation flow to use the newly introduced ProuctSelector component in this commit https://github.com/woocommerce/woocommerce-android/pull/8519/commits/eb4f8f701c0864c5f9dcbddde594267ca66796b5 of That PR. 

~~I'm trying to figure out why the tests are failing on Firebase while passing on my local machine. I'm still awaiting getting access to Firebase to investigate it myself. Just wanted to ping you so maybe you could advice something here. BTW, this PR will have smaller number of changes once the [previous one](https://github.com/woocommerce/woocommerce-android/pull/8475) gets merged.~~

EDIT: It's fixed and passing now 🟢

File: WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/tests/ui/OrdersUITest.kt [hunk 1]
```diff
@@ -27,9 +27,12 @@ class OrdersUITest : TestBase() {
     val rule = HiltAndroidRule(this)
 
     @get:Rule(order = 1)
-    val initRule = InitializationRule()
+    val composeTestRule = createAndroidComposeRule<LoginActivity>()
```
Comment (jostnes) [inline WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/tests/ui/OrdersUITest.kt:30]: When I tried this branch locally, this test failed until I updated `LoginActivity` to `MainActivity`

This was the error I got:
```
kotlin.UninitializedPropertyAccessException: lateinit property context has not been initialized
at com.woocommerce.android.AppPrefs.getPreferences(AppPrefs.kt:1006)
at com.woocommerce.android.AppPrefs.getBoolean(AppPrefs.kt:1001)
at com.woocommerce.android.AppPrefs.hasOnboardingCarouselBeenDisplayed(AppPrefs.kt:878)
at com.woocommerce.android.AppPrefsWrapper.hasOnboardingCarouselBeenDisplayed(AppPrefsWrapper.kt:200)
at com.woocommerce.android.ui.login.LoginActivity.showPrologue(LoginActivity.kt:246)
at com.woocommerce.android.ui.login.LoginActivity.onCreate(LoginActivity.kt:204)
at android.app.Activity.performCreate(Activity.java:8000)
at android.app.Activity.performCreate(Activity.java:7984)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
at androidx.test.runner.MonitoringInstrumentation.callActivityOnCreate(MonitoringInstrumentation.java:769)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3422)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3601)
at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loop(Looper.java:223)
at android.app.ActivityThread.main(ActivityThread.java:7656)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```
But it is passing on CI as is though 🤔 so not sure if it's just me...

**Review by jostnes — COMMENTED** [review]
Hey @samiuelson looks like this was merged before I got a chance to review it (I guess because the branch this was based of was merged?) I'll do a "belated" review for now using this branch: [feature/order-products-multi-select](https://github.com/woocommerce/woocommerce-android/tree/feature/order-products-multi-select). 

Overall it looks good, thanks for updating the tests to work with Compose and updating the screen name! I wasn't able to run this locally until I made one change with the Activity name, detailed comment on the affected line.

File: WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/tests/ui/OrdersUITest.kt (hunk 1 above)
Comment (samiuelson) [reply WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/tests/ui/OrdersUITest.kt:30]: Hi @jostnes, thanks for sharing the logs. I've raised a new PR with a fix for this issue: `https://github.com/woocommerce/woocommerce-android/pull/8553`.

**Pull request merged**